jdbc-v2/src/main/java/com/clickhouse/jdbc/ConnectionImpl.java

```diff
@@ -358,30 +369,49 @@ public boolean isValid(int timeout) throws SQLException {
 
     @Override
     public void setClientInfo(String name, String value) throws SQLClientInfoException {
-        throw new SQLClientInfoException("ClientInfo not supported", null);
+        try {
+            checkOpen();
+            this.defaultQuerySettings.setOption(name, value);
+        } catch (Exception e) {
+            throw new SQLClientInfoException("Failed to set client info.", Collections.singletonMap(name, ClientInfoStatus.REASON_UNKNOWN), e);
+        }
     }
 
     @Override
     public void setClientInfo(Properties properties) throws SQLClientInfoException {
-        throw new SQLClientInfoException("ClientInfo not supported", null);
+        try {
+            checkOpen();
+        } catch (SQLException e) {
+            throw new SQLClientInfoException("Failed to set client info.", new HashMap<>(), e);
+        }
+
+        for (Map.Entry<Object, Object> entry : properties.entrySet()) {
+            setClientInfo(entry.getKey().toString(), entry.getValue().toString());
+        }
     }
 
     @Override
     public String getClientInfo(String name) throws SQLException {
         checkOpen();
-        return null;
+        Object value = this.defaultQuerySettings.getAllSettings().get(name);
+        return value == null ? null : String.valueOf(value);
     }
 
     @Override
     public Properties getClientInfo() throws SQLException {
         checkOpen();
-        return new Properties();
+        Properties clientInfo = new Properties();
+        clientInfo.putAll(this.defaultQuerySettings.getAllSettings());
+        return clientInfo;
     }
 
     @Override
     public Array createArrayOf(String typeName, Object[] elements) throws SQLException {
-        //TODO: Should this be supported?
-        return null;
+        try {
+            return new com.clickhouse.jdbc.types.Array(List.of(elements));
```

Comment:
this is reference to an old project. If you need to return a simple implementation - just copy into JDBC 2 please. 
There are a lot classes so I think we need to reply on old impl as less as possible